service/history/task/transfer_active_task_executor.go

```diff
@@ -630,6 +635,25 @@ func (t *transferActiveTaskExecutor) processStartChildExecution(
 		return err
 	}
 
+	// Get target domain name
+	var targetDomainName string
+	var targetDomainEntry *cache.DomainCacheEntry
+	if targetDomainEntry, err = t.shard.GetDomainCache().GetDomainByID(task.TargetDomainID); err != nil {
+		if _, ok := err.(*types.EntityNotExistsError); !ok {
+			return err
+		}
+		// TODO: handle the case where target domain does not exist
+
+		// it is possible that the domain got deleted. Use domainID instead as this is only needed for the history event
+		targetDomainName = task.TargetDomainID
```

Comment:
any reason to assgin domain ID to domain name?

Reply:
This is the old logic for handling domain not exist case. I actually don't think it's correct as the start child request will fail anyway due to domain not exists. Since for other task types (signal and cancel) we also need to handle the domain not exists case, I left a TODO in those places and keep the old logic untouched.